metricbeat/module/kvm/dommemstat/dommemstat.go

```diff
@@ -102,44 +101,47 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 
 		c, err = net.DialTimeout(u.Scheme, address, m.Timeout)
 		if err != nil {
-			report.Error(errors.Wrapf(err, "cannot connect to %v", u))
-			return
+			return errors.Wrapf(err, "cannot connect to %v", u)
 		}
 	}
 
 	defer c.Close()
 
 	l := libvirt.New(c)
 	if err = l.Connect(); err != nil {
-		report.Error(errors.Wrap(err, "error connecting to libvirtd"))
-		return
+		return errors.Wrap(err, "error connecting to libvirtd")
 	}
 	defer func() {
 		if err = l.Disconnect(); err != nil {
-			report.Error(errors.Wrap(err, "failed to disconnect"))
+			msg := errors.Wrap(err, "failed to disconnect")
+			report.Error(msg)
+			m.Logger().Error(msg)
```

Comment:
sorry I missed previous discussions and refactor. Was report.Error logging this message before?

Reply:
@ruflin 's thinking was that `report`-ed error messages and logged error messages get sent to different places & are consumed in different ways, so we should sent errors to both.

Reply:
Thanks for explaining, LGTM